pkg/canary/util_test.go

```diff
@@ -0,0 +1,38 @@
+package canary
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestExcludeLabelsByPrefix(t *testing.T) {
+	labels := map[string]string{
+		"foo":     "bar",
+		"jenkins": "foo",
+		"flux123": "bar",
+	}
+	excludedLabelsPrefixes := []string{"jenkins", "flux"}
+
+	filteredLabels := excludeLabelsByPrefix(labels, excludedLabelsPrefixes)
+
+	assert.Equal(t, filteredLabels, map[string]string{
+		"foo": "bar",
+		// jenkins excluded
+		// and flux123 also excluded
+	})
+}
+
+func TestMakePrimaryLabels(t *testing.T) {
```

Comment:
Unrelated but I noticed this file had not unit tests so I added this one as well.